zerver/tests/test_message_topics.py

```diff
@@ -756,3 +757,39 @@ def test_get_message_edit_history(self) -> None:
         data = self.assert_json_success(result)
         self.assertEqual(data["message_history"][0]["topic"], "")
         self.assertEqual(data["message_history"][1]["prev_topic"], "")
+
+    def test_initial_state_data(self) -> None:
+        hamlet = self.example_user("hamlet")
+        iago = self.example_user("iago")
+        self.login_user(hamlet)
+
+        self.send_stream_message(hamlet, "Denmark", topic_name="")
+        self.send_stream_message(hamlet, "Verona", topic_name=Message.EMPTY_TOPIC_FALLBACK_NAME)
+
+        with mock.patch("zerver.lib.events.request_event_queue", return_value=1):
+            state_data = do_events_register(
+                iago,
+                iago.realm,
+                get_client("website"),
+                client_capabilities=ClientCapabilities(
+                    empty_topic_name=True, notification_settings_null=False
+                ),
```

Comment:
This sort of test should probably pass a short `fetch_event_types` to make the test do ~20x less work; too many `do_events_register` calls without that slows down the test suite.

Reply:
PR #34201